### PR TITLE
Add MariaDB to valid RDS engines

### PR DESCRIPTION
**Pull request by davidfischer-ch**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

rds module
##### ANSIBLE VERSION

does not apply (tested on 2.0.1.0)
##### SUMMARY

Add MariaDB to valid RDS engines.

Before:

```
fatal: [machine]: FAILED! => {"changed": false, "failed": true, "msg": "value of db_engine must be one of: MySQL,oracle-se1,oracle-se,oracle-ee,sqlserver-ee,sqlserver-se,sqlserver-ex,sqlserver-web,postgres, got: mariadb"}
```

After:

```
changed: [machine]
```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @davidfischer-ch. To the current maintainers, @bpennypacker, @willthames please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by ryansb** [comment]
:+1: LGTM

**Comment by gregdek** [comment]
Thanks again to @davidfischer-ch for this PR, and thanks @bpennypacker, @willthames for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @davidfischer-ch for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by davidfischer-ch** [comment]
ready_for_review

**Comment by ryansb** [comment]
Simple rebase. :+1: shipit

**Comment by gregdek** [comment]
Thanks again to @davidfischer-ch for this PR, and thanks @bpennypacker, @willthames for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/amazon/rds.py [hunk 1]
```diff
@@ -42,7 +42,7 @@
       - The type of database.  Used only when command=create.
```
Comment (resmo) [inline cloud/amazon/rds.py:42]: Please add a little note in the docs e.g.

```
- mariadb was added in version 2.2
```

Comment (davidfischer-ch) [inline cloud/amazon/rds.py:42]: Hi, can anyone more experienced than me can handle this please?
I am not sure where and how to write this note.

Thanks.

Comment (resmo) [inline cloud/amazon/rds.py:42]: No problem, sorry for not made it clear:

The documentation should look like:

``` yaml
  db_engine:
    description:
      - The type of database. Used only when command=create.
      - mariadb was added in version 2.2
    required: false
    default: null
    choices: [ 'mariadb', 'MySQL', 'oracle-se1', 'oracle-se', 'oracle-ee', 'sqlserver-ee', 'sqlserver-se', 'sqlserver-ex', 
```

Comment (davidfischer-ch) [inline cloud/amazon/rds.py:42]: Done!

**Pull request merged**